cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ApplicationRequestsPlaner.java

```diff
@@ -1,164 +1,273 @@
 package cloud.foundry.cli.logic.apply;
 
-import static com.google.common.base.Preconditions.*;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 import cloud.foundry.cli.crosscutting.exceptions.ApplyException;
 import cloud.foundry.cli.crosscutting.logging.Log;
 import cloud.foundry.cli.crosscutting.mapping.beans.ApplicationBean;
+import cloud.foundry.cli.crosscutting.mapping.beans.ApplicationManifestBean;
+import cloud.foundry.cli.crosscutting.mapping.validation.Field;
+import cloud.foundry.cli.crosscutting.mapping.validation.ObjectPropertyValidation;
+import cloud.foundry.cli.crosscutting.mapping.validation.ScalarField;
 import cloud.foundry.cli.logic.diff.change.CfChange;
+import cloud.foundry.cli.logic.diff.change.ChangeType;
+import cloud.foundry.cli.logic.diff.change.container.CfContainerChange;
+import cloud.foundry.cli.logic.diff.change.container.CfContainerValueChanged;
+import cloud.foundry.cli.logic.diff.change.map.CfMapChange;
+import cloud.foundry.cli.logic.diff.change.map.CfMapValueChanged;
 import cloud.foundry.cli.logic.diff.change.object.CfNewObject;
-import cloud.foundry.cli.logic.diff.change.object.CfObjectValueChanged;
 import cloud.foundry.cli.logic.diff.change.object.CfRemovedObject;
 import cloud.foundry.cli.operations.ApplicationsOperations;
+import org.cloudfoundry.operations.applications.ApplicationHealthCheck;
+import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
-
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import reactor.core.publisher.Mono;
 
 import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Predicate;
 
 /**
  * This class is responsible to build the requests in the context of
  * applications according to the CfChanges. The class does create the request
  * tasks by implementing the {@link CfChangeVisitor} interface.
  */
-public class ApplicationRequestsPlaner extends RequestsPlaner {
+public class ApplicationRequestsPlaner {
 
     private static final Log log = Log.getLog(ApplicationRequestsPlaner.class);
-    private static final Set<String> FIELDS_REQUIRE_RESTART = new HashSet<>(Arrays.asList("meta", "path"));
-
-    private final ApplicationsOperations appOperations;
-    private final String applicationName;
 
-    private ApplicationRequestsPlaner(ApplicationsOperations appOperations, String applicationName) {
-        this.appOperations = appOperations;
-        this.applicationName = applicationName;
-    }
+    private static final Map<String, Field> FIELDS_REQUIRE_RESTART = new HashMap<String, Field>() {{
```

Comment:
To explain this data structure:
```suggestion
    // stores field-names related to applications that require an app restart when their values in the configuration are changed
    // their corresponding Field-instances are only used for validation purposes
    private static final Map<String, Field> FIELDS_REQUIRE_RESTART = new HashMap<String, Field>() {{
```